### PR TITLE
fix: glitch on Popover with an hover

### LINKED ISSUE #2246: Popover : triggerMethod hover
### [Design] For which Figma library?

None

### For which component/foundations?

Popover

### Prioritization

High 🌋

### Current Behavior

Popover with `triggerMethod = hover` has a flaky behavior when other popovers are in the DOM (see video).

### Expected Behavior

Popover should appear and disappear independently from the presence of other tooltip

### Screenshots / Videos

https://github.com/WTTJ/welcome-ui/assets/33298885/502fd75b-43d2-41a1-b8db-29328a455a15

### [Tech] Welcome UI Version

_No response_

### [Tech] Browser

_No response_

### [Tech] Operating System

None

### PR DESCRIPTION
**Pull request by theo-mesnil**
Fixes #2246 

Because of glitch on hover on the Popover due to a hack on our side, we rework this component to have now 2 options:
- Popover (like today)
- PopoverHover (it's Hovercard from Ariakit)

We delete the property `triggerMethod`:

```diff
+ import { usePopoverHover, PopoverHover } from '@welcome-ui/popover'

function() {
- const popover = usePopover({ triggerMethod: 'hover' })
+ const popover = usePopoverHover()

  return (
    <>
-    <Popover.Trigger as={Button} store={popover}>
+    <PopoverHover.Trigger as={Button} store={popover}>
        ...
-     </Popover.Trigger>
+     </PopoverHover.Trigger>
-     <Popover store={popover}>
+     <PopoverHover store={popover}>
          ...
-     </Popover>
+     </PopoverHover>
    </>
  )
}
```

https://github.com/WTTJ/welcome-ui/assets/50322149/dad9527e-af76-484b-a303-5f8cca447ff5

### CODE REVIEW

File: packages/Popover/src/Arrow.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+
+import { UsePopover } from './usePopover'
+import * as S from './styles'
+
+const transformMap: { [key: string]: string } = {
```
Comment (mleralec) [inline packages/Popover/src/Arrow.tsx:6]: je pense qu'on peut supprimer le type ici et laisser typescript le faire tout seul

File: packages/Popover/src/Content.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { PopoverProps } from './Popover'
+
+export interface ContentOptions {
+  children: PopoverProps['children']
+  /** call a function when popover closed */
+  onClose?: () => void
+  store: UsePopover | UsePopoverHover
+}
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (mleralec) [inline packages/Popover/src/Content.tsx:18]: eslint comment is useless?

File: packages/Popover/src/Popover.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+export interface PopoverOptions extends Ariakit.PopoverProps {
+  /** call a function when popover closed */
+  onClose?: () => void
+  store: UsePopover
+}
+
+export type PopoverProps = CreateWuiProps<'div', PopoverOptions>
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (mleralec) [inline packages/Popover/src/Popover.tsx:18]: same here

File: packages/Popover/src/PopoverHover.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+export interface PopoverHoverOptions extends Ariakit.HovercardProps {
+  /** call a function when popover closed */
+  onClose?: () => void
+  store: UsePopoverHover
+}
+
+export type PopoverHoverProps = CreateWuiProps<'div', PopoverHoverOptions>
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (mleralec) [inline packages/Popover/src/PopoverHover.tsx:18]: same here

**Review by mleralec — COMMENTED** [review]
So much simpler 👍🏻

**Review by RobelTekle — APPROVED** [review]
🚀

**Pull request merged**